builtin/k8s/platform.go

```diff
@@ -125,6 +128,164 @@ func (p *Platform) getClientset() (*clientsetInfo, error) {
+
+		var mostRecentCondition v1.DeploymentCondition
+		for _, condition := range deployResp.Status.Conditions {
+			if condition.LastUpdateTime.Time.After(mostRecentCondition.LastUpdateTime.Time) {
+				mostRecentCondition = condition
+			}
+		}
+
+		var deployHealth sdk.StatusReport_Health
+		switch mostRecentCondition.Type {
+		case v1.DeploymentAvailable:
+			deployHealth = sdk.StatusReport_READY
+		case v1.DeploymentProgressing:
+			deployHealth = sdk.StatusReport_ALIVE
+		case v1.DeploymentReplicaFailure:
+			deployHealth = sdk.StatusReport_DOWN
+		default:
+			deployHealth = sdk.StatusReport_UNKNOWN
+		}
+
+		// Redact env vars from containers - they can contain secrets
+		for i := 0; i < len(deployResp.Spec.Template.Spec.Containers); i++ {
+			deployResp.Spec.Template.Spec.Containers[i].Env = []corev1.EnvVar{}
+		}
+
+		deployStateJson, err := json.Marshal(map[string]interface{}{
+			"deployment": deployResp,
+		})
+		if err != nil {
+			return status.Errorf(codes.FailedPrecondition, "failed to marshal deployment to json: %s", err)
+		}
+
+		deploymentResource.Name = deployResp.Name
+		deploymentResource.Id = fmt.Sprintf("%s", deployResp.UID)
+		deploymentResource.CreatedTime = timestamppb.New(deployResp.CreationTimestamp.Time)
+		deploymentResource.Health = deployHealth
+		deploymentResource.HealthMessage = fmt.Sprintf("%s: %s", mostRecentCondition.Type, mostRecentCondition.Message)
+		deploymentResource.StateJson = string(deployStateJson)
+	}
+
+	// Get pod status
+
+	podClient := clientset.Clientset.CoreV1().Pods(namespace)
+	podLabelId := fmt.Sprintf("app=%s", deploymentState.Name)
+	podList, err := podClient.List(ctx, metav1.ListOptions{LabelSelector: podLabelId})
+	if err != nil {
+		return status.Errorf(codes.FailedPrecondition, "error listing pods to determine application health: %s", err)
+	}
+	if podList == nil {
+		return status.Errorf(codes.FailedPrecondition, "kubernetes pod list cannot be nil")
+	}
+
+	for _, pod := range podList.Items {
+		// Redact env vars because they can contain secrets
+		for i := 0; i < len(pod.Spec.Containers); i++ {
+			pod.Spec.Containers[i].Env = []corev1.EnvVar{}
+		}
+
+		podJson, err := json.Marshal(map[string]interface{}{
+			"pod":       pod,
+			"hostIP":    pod.Status.HostIP,
+			"ipAddress": pod.Status.PodIP,
+		})
+		if err != nil {
+			return status.Errorf(codes.Internal, "failed to marshal k8s pod definition to json: %s", podJson)
+		}
+
+		var health sdk.StatusReport_Health
+		var healthMessage string
+
+	OUTER:
+		switch pod.Status.Phase {
+		case corev1.PodPending:
+			health = sdk.StatusReport_ALIVE
+		case corev1.PodRunning:
+			// Extra checks on the latest condition to ensure pod is reporting ready and running
+			for _, c := range pod.Status.Conditions {
+				if c.Status == corev1.ConditionTrue && c.Type == corev1.PodReady {
+					health = sdk.StatusReport_READY
+					healthMessage = fmt.Sprintf("ready")
+					break OUTER
```

Comment:
Do we need to go back or can we just `break` out of the loop here and keep going?

Reply:
You had a good idea for how to avoid this label, and I fixed it.